### DISCUSSION TITLE
Nexus - Cursor pagination in nested field

### DISCUSSION THREAD

**Discussion by Miigaarino** [prisma/prisma#16045] [Q&A] [answered]
Hello, I have below tables with many-to-many relationship:
```
model User {
    ...
    projects      Project[]
    ...
}

model Project {
    ...
    creators       User[]
    ....
}

model Tag {
    ...
    projects      Project[]
    ...
}

```

I want cursor paginated results on projects field on every table that has projects:
- When getting single user data get only first N projects
- When getting single tag data get only first N projects

**What approach should I use for this kind of result?**

Current solution: 
- Tag type (pagination defined in projects field resolver):
```ts
export const Tag = objectType({
  name: "Tag",
  definition(t) {
    ...
    t.field("projects", {
      type: ProjectsPaginated,
      args: {
        filter: PaginationFilterInput,
      },
      async resolve(parent, { filter }, ctx) {
        const id = filter?.cursor;
        const search = filter?.search;
        const mostDownloadedOnTop = filter?.isTopDownloaded;

        const items = await ctx.prisma.project.findMany({
          where: {
            name: { contains: search as string, mode: "insensitive" },
            tags: {
              some: {
                id: parent.id,
              },
            },
          },
          skip: id ? 1 : 0,
          cursor: id
            ? {
                id,
              }
            : undefined,
          take: PAGE_SIZE,
          orderBy: mostDownloadedOnTop
            ? [{ download_count: "desc" }, { created_at: "desc" }]
            : { created_at: "desc" },
        });

        const lastProjectInResults = items[PAGE_SIZE - 1];

        return {
          items,
          cursor: lastProjectInResults?.id,
        };
      },
    });
  },
});
```

- User type (projects resolver only returns parents projects, will have to implement above cursor pagination in query resolver):
```ts
export const User = objectType({
  name: "User",
  definition(t) {
    ...
    t.list.field("projects", {
      type: Project,
      async resolve(parent, _args, ctx) {
        return ctx.prisma.user
          .findUnique({
            where: {
              id: parent.id,
            },
          })
          .projects();
      },
    });
  },
});
```

The question is which of the above solutions should I use? 
- get many-to-many relationships using findMany, some (can write the pagination in type definer) approach 

**OR** 
- findUnique, select (will write the pagination in query resolver) approach?

**Or are there any other options that you think I should use?**

**Comment by pantharshit00** [comment, marked as answer]
Hey @Miigaarino 

Can you please share a small reproduction of the structure in a GIt repo? It will help us in answering this more accurately.

From what I can infer from the current information, `get many-to-many relationships using findMany, some (can write the pagination in type definer) approach` would be better as there will no need for a joining relation data when Prisma will try to resolve the query as which be the case for the other one.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
